libs/auth/react/src/lib/auth/AuthProvider.tsx

```diff
@@ -207,18 +216,17 @@ export const AuthProvider = ({
     }
   }, [dispatch, userManager, signIn, autoLogin, state.userInfo === null])
 
-  const isCurrentRoute = (path?: string) =>
-    isDefined(path) && url.includes(basePath + path)
-
   const init = async () => {
-    if (isCurrentRoute(authSettings?.redirectPath)) {
+    const currentUrl = getCurrentUrl(basePath)
+
+    if (isCurrentRoute(currentUrl, authSettings.redirectPath)) {
       try {
         const user = await userManager.signinRedirectCallback(
           window.location.href,
         )
 
-        const url = typeof user.state === 'string' ? user.state : '/'
-        window.history.replaceState(null, '', url)
+        const url = typeof user.state === 'string' ? user.state : basePath
```

Comment:
This may redirect to `basePath + basePath`

```suggestion
        const url = typeof user.state === 'string' ? user.state : '/'
```